internal/project/project.go

```diff
@@ -385,51 +382,37 @@ func (p *Project) markFileAsDirty(path tspath.Path) {
 }
 
 func (p *Project) markAsDirty() {
-	p.dirtyStateMu.Lock()
-	defer p.dirtyStateMu.Unlock()
+	p.mu.Lock()
+	defer p.mu.Unlock()
 	p.dirtyFilePath = ""
 	if !p.dirty {
 		p.dirty = true
 		p.version++
 	}
 }
 
-// updateIfDirty returns true if the project was updated.
-func (p *Project) updateIfDirty() bool {
-	// !!! p.invalidateResolutionsOfFailedLookupLocations()
-	return p.dirty && p.updateGraph()
-}
-
-func (p *Project) onFileAddedOrRemoved(isSymlink bool) {
-	p.dirtyStateMu.Lock()
-	defer p.dirtyStateMu.Unlock()
-	p.hasAddedOrRemovedFiles = true
```

Comment:
These weren’t yet used for anything but logging, and they complicate the locking because they are sometimes, but not always, called during program construction when the mutex is already locked.